File: Ex02/Animals_ObjC/Reptile.m
//
//  Reptile.m
//  
//
//  Created by Elisabeth Hoppe on 25.05.15.
//
//

#import "Reptile.h"
@implementation Reptile:Animal

-(id)initWithParams:(NSMutableString*)name_ carnivore:(BOOL)carnivore_ bodyTemperature:(double)bodyTemperature_
{
    self=[super initWithParams: name_ carnivore:carnivore_];
    
    if(self)
    {
        
        [self setBodyTemperature:bodyTemperature_];
    }

    return self;
}

-(void)setBodyTemperature: (double)bodyTemperature_{
    bodyTemperature = bodyTemperature_;
}

-(double)getBodyTemperature{
    return bodyTemperature;
}

-(void)toString
{
    //TODO
}

-(BOOL)equals:(Reptile*)reptile
{
    
    //check the super class equality
    BOOL animalEqualiy = [super equals:self];
    
    if(animalEqualiy == YES)
    {
        //check for equality in own class
        if(bodyTemperature == [reptile getBodyTemperature])
        {
            return YES;
        }
        else
        {
            return NO;
        }
    }
    else
    {
        return NO;
    }
    
    return NO;
}

@end

File: Ex02/Animals_JS/js/Reptile.js
var Reptile = function(name, carnivore, bodyTemperature){
	//calling super constructor
	Animal.call(this, name, carnivore);
	
	this.setBodyTemperature = function(bodyTemperature){
	
		this.pBodyTemperature = bodyTemperature;
	}

	this.getBodyTemperature = function(){
		return this.pBodyTemperature;

	}
	
	//bodyTemperature ist privat
	var pBodyTemperature = this.setBodyTemperature(bodyTemperature);
	
	
	//da abstrakte Klasse
	if(this.constructor === Reptile){
		throw new Error ("Reptile: Cannot instantiate abstract class");
		// console.log("Reptile: Cannot instatiate abstract class");
	}
	
	
};

Reptile.prototype = Object.create(Animal.prototype);
Reptile.prototype.constructor = Reptile;

//Elternklasse-Methoden




Reptile.prototype.equals = function(reptileOb){
	
	var animalEqality = Object.getPrototypeOf(Reptile.prototype).equals(reptileOb);
	
	if(animalEquality == true)
	{
		if( (this.getBodyTemperature() == reptileOb.getBodyTemperature()) )
		{
			return true;
		}
		return false;
	}
	
	else
	{
		return false;
	}
	
	
 }



function createReptile() {
	try{

		console.log("creating reptile...");
		var m = new Reptile ("Test" , "true");
		return m;
	}
	catch(e){
	
		console.log("Exception:" + e);
		return e;
	}
};